src/screens/expandedAssetSheet/components/sections/HistorySection.tsx

```diff
@@ -0,0 +1,335 @@
+    const rowHeightWithPadding = ROW_HEIGHT + ROW_PADDING;
+    const amount = isExpanded ? interactions.length : minOfTwoOptions;
+
+    const rowHeight = mulWorklet(amount, rowHeightWithPadding);
+    return Number(ceilWorklet(sumWorklet(rowHeight, moreButtonHeight)));
+  }, []);
+
+  const containerStyles = useAnimatedStyle(() => {
+    const height = calculateContainerHeight(filteredTokenInteractions, isExpanded.value);
+    return {
+      height: withSpring(clamp(height, 0, height), SPRING_CONFIGS.springConfig),
+    };
+  });
+
+  const loaderStyles = useAnimatedStyle(() => {
+    const shouldShowMoreButton = filteredTokenInteractions.length > DEFAULT_VISIBLE_ITEM_COUNT;
+    const moreButtonHeight = shouldShowMoreButton ? MORE_BUTTON_HEIGHT + LIST_BUTTON_GAP : 0;
+    const loaderHeight =
+      DEFAULT_VISIBLE_ITEM_COUNT * ROW_HEIGHT + (DEFAULT_VISIBLE_ITEM_COUNT - 1) * ROW_PADDING + LIST_PADDING * 2 + moreButtonHeight;
+
+    return {
+      height: withTiming(isLoading && filteredTokenInteractions.length === 0 ? loaderHeight : 0, TIMING_CONFIGS.fastFadeConfig),
+      opacity: withTiming(isLoading && filteredTokenInteractions.length === 0 ? 1 : 0, TIMING_CONFIGS.fastFadeConfig),
+    };
+  });
+
+  const listStyles = useAnimatedStyle(() => {
+    return {
+      opacity: withTiming(filteredTokenInteractions.length !== 0 ? 1 : 0, TIMING_CONFIGS.fastFadeConfig),
+    };
+  });
+
+  const renderItem = useCallback(
+    ({ item }: { item: TokenInteraction }) => {
+      return <ListItem item={item} nativeCurrency={nativeCurrency} />;
+    },
+    [nativeCurrency]
+  );
+
+  return (
+    <Box as={Animated.View} style={[containerStyles, { position: 'relative' }]}>
+      <Box as={Animated.View} style={loaderStyles} sharedTransitionTag="history-section" sharedTransitionStyle={historyTransition} gap={24}>
+        {Array.from({ length: DEFAULT_VISIBLE_ITEM_COUNT }).map((_, index) => (
+          <SkeletonRow key={index} width={DEVICE_WIDTH - 48} height={ROW_HEIGHT} />
+        ))}
+      </Box>
+
+      <Animated.FlatList
+        sharedTransitionTag="history-section"
+        sharedTransitionStyle={historyTransition}
+        data={filteredTokenInteractions}
+        style={[listStyles, { paddingVertical: LIST_PADDING }]}
+        contentContainerStyle={{ gap: 24 }}
+        scrollEnabled={false}
+        renderItem={renderItem}
+      />
+      <MoreButton tokenInteractions={filteredTokenInteractions} />
+    </Box>
+  );
+});
+
+export const HistorySection = memo(function HistorySection() {
+  const { accentColors, tokenInteractions, isLoadingTokenInteractions } = useExpandedAssetSheetContext();
+
+  const buys = useMemo(() => {
+    return tokenInteractions?.filter(interaction => interaction.direction === TokenInteractionDirection.In);
+  }, [tokenInteractions]);
+
+  const sells = useMemo(() => {
+    return tokenInteractions?.filter(interaction => interaction.direction === TokenInteractionDirection.Out);
+  }, [tokenInteractions]);
```

Comment:
These don't just include buys and sells but buys & receives and sells & sents

Reply:
I think that grouping is intentional, no? We can't fit 4 tabs in this area unless we want to do a horizontal scroll